### PR TITLE
LLM execution thread notifications and error handling 

### PR DESCRIPTION
**Pull request by mzegla**
### 🛠 Summary

JIRA/Issue if applicable.
Describe the changes.

### 🧪 Checklist

- [ ] <s>Unit tests added.</s> Tests will be added after passing sampling params is supported 
- [ ] <s>The documentation updated.</s> Documentation is not in place yet
- [x] Change follows security best practices.
``

### CODE REVIEW

File: src/llm/llm_executor.hpp [hunk 1]
```diff
@@ -17,32 +17,78 @@
...
-    static void run(std::shared_ptr<ContinuousBatchingPipeline> pipe, std::atomic<bool>* end) {
-        while (!(*end)) {
-            pipe->step();
+    static void run(LLMExecutor* llmExecutor, std::atomic<bool>* receivedEndSignal) {
+        while (!(*receivedEndSignal)) {
+            try {
+                if (llmExecutor->hasRequests()) {
+                    llmExecutor->step();
+                } else {
+                    llmExecutor->waitForRequests(receivedEndSignal);
```
[the remark this answers is not in the document]
Comment (dkalinowski) [reply src/llm/llm_executor.hpp:70]: @mzegla

[the remark this answers is not in the document]
Comment (mzegla) [reply src/llm/llm_executor.hpp:70]: No, see the predicate in the wait() call for conditional variable inside waitForRequests method. 
We pass `receivedEndSignal` exactly for that purpose. When we want to finish the thread we set that flag and notify CV so it breaks out of this wait.

**Review by rasapala — APPROVED** [review]
LGTM

File: src/test/llmnode_test.cpp [hunk 2]
```diff
@@ -122,7 +131,7 @@ TEST_F(LLMFlowKfsTest, LLMNodeNameMissing) {
             output_stream: "RESPONSE:out2"
             node_options: {
                 [type.googleapis.com / mediapipe.LLMCalculatorOptions]: {
-                    workspace_path: "/workspace/"
+                    models_path: "/workspace/"
```
Comment (dtrawins) [inline src/test/llmnode_test.cpp:134]: later, also relative path should be supported...

**Review by dtrawins — APPROVED** [review]

File: src/test/llmnode_test.cpp (hunk 2 above)
Comment (mzegla) [reply src/test/llmnode_test.cpp:134]: @rasapala I think when you added node initialization you take that into account. Can you confirm it should work?

**Review by dkalinowski — APPROVED** [review]

File: src/test/llmnode_test.cpp (hunk 2 above)
Comment (rasapala) [reply src/test/llmnode_test.cpp:134]: Yeah it should work, but I have not tested it yet. Maybe with the model tests we can add test for relative? Or next PR.

**Pull request merged**